### ISSUE TITLE
How to access configuration?

### ISSUE THREAD

**Issue by Lesani** [JoDehli/PyLoxone#145]
I'm trying to access the configuration as seen here:
![image](https://user-images.githubusercontent.com/5182104/203183942-833171a7-113c-4553-9a1c-c11b205eda84.png)
But for the life of me I cannot find how I can access that, in order to configure my loxone settings...

I have clicked every possible button, 5 times over I think... how do I get there?

I installed pyloxone via hacs, and it is visible in my hacs integrations. I tried the latest 3.xx and the 4.0 version

**Comment by JoDehli** [comment]
Have you added the pyloxone in your integrations? First you must install it via hacs as you did it. Then you must add the integration on the integration tab ( settings -> devices&services -> integration tab). Did you do this?

**Comment by Lesani** [comment]
Hello! Yes I have added it there https://i.imgur.com/ZM9sCCu.png

**Comment by JoDehli** [comment]
Habe you access to your installation via samba? If yes you can check if a installation folder exists under custom_components.

**Comment by Lesani** [comment]
I have found the settings, but have no luch setting up my Miniserver Gen2.

I am running version 0.4.0 of your the integration
I have tried ports 80, 8080, and 443
with & without the checkboxes for scenes and lightcontroller
I have triple verified my address and password

https://i.imgur.com/7tJepKU.png

`Error setting up entry PyLoxone for loxone
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 365, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/loxone/__init__.py", line 195, in async_setup_entry
    if not await miniserver.async_setup():
  File "/config/custom_components/loxone/miniserver.py", line 140, in async_setup
    request_code = await self.lox_config.getJson()
  File "/config/custom_components/loxone/api.py", line 122, in getJson
    api_resp = await client.get("/jdev/cfg/apiKey")
  File "/usr/local/lib/python3.10/site-packages/httpx/_client.py", line 1751, in get
    return await self.request(
  File "/usr/local/lib/python3.10/site-packages/httpx/_client.py", line 1527, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
  File "/usr/local/lib/python3.10/site-packages/httpx/_client.py", line 1614, in send
    response = await self._send_handling_auth(
  File "/usr/local/lib/python3.10/site-packages/httpx/_client.py", line 1642, in _send_handling_auth
    response = await self._send_handling_redirects(
  File "/usr/local/lib/python3.10/site-packages/httpx/_client.py", line 1700, in _send_handling_redirects
    raise exc
  File "/usr/local/lib/python3.10/site-packages/httpx/_client.py", line 1682, in _send_handling_redirects
    await hook(response)
  File "/config/custom_components/loxone/api.py", line 84, in raise_if_not_200
    raise LoxoneHTTPStatusError(
custom_components.loxone.api.LoxoneHTTPStatusError: Miniserver response code 500`

**Comment by JoDehli** [comment]
Ok the problem is only local access is possible. Please search in the closed issues for similar problems. Local acces can be set in the loxone config.

**Comment by Lesani** [comment]
The problem is that local acces is already allowed on my miniserver but your manual/screenshot is wrong. you don't add your loxone cloud dns address, but the "local" dns cloud address - for me that is IP.MAC.dnyndns.loxonecloud.com, this address is generated in the browser window automatically when you access the local IP of the miniserver
10-0-0-24.xxxxxxxxxx.dyndns.loxonecloud.com

**Comment by JoDehli** [comment]
Does this mean now it is working?

**Comment by Lesani** [comment]
yes with this address it connected successfully - however I cannot control anything yet. It pulled all my devices and shows them on the dashboard, but for example pussing the "blinds up" button does nothing.

**Comment by JoDehli** [comment]
Any logs?

**Comment by JoDehli** [comment]
And please paste the screenshots direct into git  comment.

**Comment by Lesani** [comment]
deleted the integration, and re-added it with both checkboxes checked (scenes and lightcontroller) now some things work.

I can get the blinds to go up or stop moving, but going down is greyed out. all other switches do nothing, and presence is not shown.

![image](https://user-images.githubusercontent.com/5182104/204088406-748eede1-73ed-47db-b88b-5de14f596c2f.png)

The usecase I need this for is to get other sensors - e.g. philips hue motion sensors - to trigger things in loxone, like presence in a room. Can that actually be done with this yet?

**Comment by JoDehli** [comment]
Greyed out means that there are already down. Of course these devices work https://github.com/JoDehli/PyLoxone#supported-loxone-entites

**Comment by Lesani** [comment]
they are not fully down, the blinds are not closed, they are on shade (hence 99%) - and to test that I moved them up fully using the button in HA, the "down" stayed greyed out even when the blinds were fully up

plus they stopped reacting again now, nothing happens on any button.
I deleted the integration again, and re-added it, now the blinds work again.

Where can I find the logs you referred to earlier? maybe something is visible there

**Comment by Lesani** [comment]
and they stopped working again - settings - system - logs gives me this:

`Logger: homeassistant
Source: custom_components/loxone/api.py:386
Integration: PyLoxone ([documentation](https://github.com/JoDehli/PyLoxone), [issues](https://github.com/JoDehli/PyLoxone/issues))
First occurred: 1:33:32 PM (6 occurrences)
Last logged: 1:33:33 PM

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/websockets/legacy/protocol.py", line 979, in transfer_data
    await asyncio.shield(self._put_message_waiter)
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/loxone/miniserver.py", line 189, in listen_loxone_send
    await self.api.send_websocket_command(device_uuid, value)
  File "/config/custom_components/loxone/api.py", line 386, in send_websocket_command
    await self._ws.send(command)
  File "/usr/local/lib/python3.10/site-packages/websockets/legacy/protocol.py", line 635, in send
    await self.ensure_open()
  File "/usr/local/lib/python3.10/site-packages/websockets/legacy/protocol.py", line 944, in ensure_open
    raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedError: sent 1011 (unexpected error) keepalive ping timeout; no close frame received`

**Comment by JoDehli** [comment]
I think you configuration of the url is not supported by pyloxone. I have only an gen1 and can improve it for the gen2. Sorry.

**Comment by Lesani** [comment]
ok, wird as it works half-way for a minute or two. I just sold my Gen1 for about the same price on ebay as a new gen2 costs, I switched because Gen1 web interface does not support intercom live streaming. I'll try my luck with loxberry (having problems there as well) and check back here every so often, maybe at some point you manage to interface with the Gen2. Seeing as gen1 is a discontinued product, maybe someone forks it and modifies it ;)

**Comment by JoDehli** [comment]
I already implemented in the newer version of pyloxone but there I have a other problem. I know users which are using the gen2 successfully. It is only the local access url which is wrong. 

But good luck with the loxberry.

**Comment by Lesani** [comment]
I don't have any other "local access" url?

I tried it now again with my direct IP address (10.0.0.24) and instead of giving me an error it also works for approx 1 minute to send the commands for the blinds. It does not receive any data from loxone though, the presence is perma-off and also the position of the blinds is fixed at "closed", even though they are moving after pressing the "up" button. So it seems pyloxone can send data to my gen2 for a mintute, but does not receive any stats from anything (presence, blind position, humidity, etc)

**Comment by JoDehli** [comment]
If you use the local ip do not use port 80. 

I am sure it is a thing of a few minutes to get it to work. But I have no access to a loxone gen2. I know that there is a test server from loxone but it is a gen1. Have you tried to get nodered to work. 

It is only a small change. The problem is that the gen2 (and of course gen1) are to weak for a real ssl encryption. And that makes more complicated than it should be. 

If you want you can join the discord server there is a guy how already uses a gen2 with pyloxone.

**Comment by JoDehli** [comment]
https://discord.gg/vXJ2MZka here is the invitation.

**Comment by Lesani** [comment]
Hi!
Can you send another invite link? I wasn't at home and it expired by now.
Thanks!

**Comment by Lesani** [comment]
also: sometimes the integration already works, I had it running for half an hour, with a hue motion sensor switching a switch in loxone (that in turn is connected to a buzzer)... it worked for half an hour, then it stopped again. I need to restart hass completely or uninstall/reinstall the plugin to get it working again. I strongly disaggree that it has generally to do with the gen2 server or such, otherwise it wouldn't work for [insert random amount] minutes before it stops working. if the url was bad it wouldn't work at all

**Comment by JoDehli** [comment]
hey come check out Discord with me https://discord.gg/Pd8Cqsx6

**Comment by JoDehli** [comment]
Do you use any custom entities in your setup? I had problems with custom entries togehter with  pyloxone. Does pyloxone stop working when you switch something? I mean can you force the error.

**Comment by Lesani** [comment]
Sorry I don't know what a "custom entry" is, I am a HASS noob, I only have one set up right now for PyLoxone.
-I have the "Mosquitto broker" (for possibly avoiding PyLoxone and try and go via Loxberry, also thus far unsuccessfull
-Studio Code Server, to edit files, this I only added 2 days ago, I have not edited any files yet
-SSH and HACS

I can actually not force the error anymore. The only thing I did from the times where I could only operate the blinds and lights from the dashboard for ~1minute to now is:
- install Visual Studio Code Server
- Add 2 Automations
![image](https://user-images.githubusercontent.com/5182104/206858166-03eac820-18dd-4173-9cb0-045cce3df7d8.png)
![image](https://user-images.githubusercontent.com/5182104/206858189-5edba091-e593-4851-9bb5-e1433a210744.png)

I do not know why, I do not know how, but somewhere in those 2 actions the issue resolved itself. Where before - see messages above - I could not operate the "blinds down" button - because PyLoxone "thought" they were all the way down even though they werent, now there is actual correct feedback on their position and I can operate them up and down as appropriate. So I guess my issue is solved for now, even though very unsatisfyingly. Anyone else who might have that issue can only try to repeat my steps.

**Comment by Lesani** [comment]
I have recreated the problematic state. I deleted the integration in an effort to to refresh my loxone names (I had multiple entities named the same name, in different rooms, but e.g. when building a dashboard it does not show the room, so I renamed everything uniquely in Loxone.
After re-adding the integration I again had errors, tried with different addresses, fixed IP, the Loxone DNS Address, nothing worked until suddenly it did with the loxone dns. All entities are now again discovered, but I have the same problematic state that - e.g. for the blinds, they show as down
I can push the "up" button -> they go up
I can push the stop button -> they stop
The "down" button stays greyed out, and everything stops working after a minute.
